PsfLauncher/main.cpp

```diff
@@ -230,6 +244,170 @@ void GetAndLaunchMonitor(const psf::json_object& monitor, std::filesystem::path
     LaunchMonitorInBackground(packageRoot, monitorExecutable->as_string().wide(), monitorArguments->as_string().wide(), wait, asAdmin, cmdShow, dirStr);
 }
 
+void LaunchInBackgroundAsAdmin(const wchar_t executable[], const wchar_t arguments[], bool wait, int cmdShow, LPCWSTR dirStr)
+{
+    std::wstring cmd = L"\"";
+    cmd.append(executable);
+    cmd.append(L"\"");
+
+
+    // This happens when the program is requested for elevation.  For now just launch this way.  We don't get to inject,
+    // but it is better than nothing.
+    SHELLEXECUTEINFOW shExInfo =
+    {
+        sizeof(shExInfo) // bSize
+        , wait ? (ULONG)SEE_MASK_NOCLOSEPROCESS : (ULONG)(SEE_MASK_NOCLOSEPROCESS | SEE_MASK_WAITFORINPUTIDLE) // fmask
+        , 0           // hwnd
+        , L"runas"    // lpVerb
+        , cmd.c_str() // lpFile
+        , arguments   // lpParameters
+        , dirStr     // lpDirectory
+        , cmdShow           // nShow
+        , 0           // hInstApp
+    };
+    Log("PsfLaunch: Unable to launch with injection (possibly due to elevation).  Launch using shell launch without injection.");
+    THROW_LAST_ERROR_IF_MSG(!ShellExecuteEx(&shExInfo), "Error starting monitor using ShellExecuteEx");
+    THROW_HR_IF(HRESULT_FROM_WIN32(ERROR_INVALID_HANDLE), shExInfo.hProcess == INVALID_HANDLE_VALUE);
+
+    // possibly inject the runtime now?
+    try
+    {
+        BOOL b;

```

Comment:
Do not use single letter variable names.

Reply:
Rectified the variable name and renames as isProcessInJob